app/celery/scheduled_tasks.py

```diff
@@ -511,12 +511,18 @@ def letter_raise_alert_if_no_ack_file_for_zip():
             for zip_file in content.split('\n'):    # each line
                 s = zip_file.split('|')
                 for zf in zip_file_list:
-                    if s[0].lower() in zf.lower():
+                    if s[0].upper() in zf:
                         zip_file_list.remove(zf)
```

Comment:
you shouldn't mutate state of a list while traversing it - see https://unspecified.wordpress.com/2009/02/12/thou-shalt-not-modify-a-list-during-iteration/

what's probably better, is to create a `zip_file_set`, and an `ack_file_set`, and then just assert that they are equal

Reply:
```
zip_file_set = {
    key.upper().rstrip('.TXT') 
    for key in s3.get_list_of_files_by_suffix(
        bucket_name=current_app.config['LETTERS_PDF_BUCKET_NAME'],
        subfolder=datetime.utcnow().strftime('%Y-%m-%d') + '/zips_sent',
        suffix='.TXT'
    )
}

ack_file_set = set()

for key in ack_file_list....
    ...
          ack_file_set.add(s[0].upper())

if ack_file_set != zip_file_set:
    logger.error('Ack file set: {}, Zip file set: {}'.format(ack_file_set, zip_file_set))
    deskpro_client.create_ticket(message='ack file doesn't match what we sent')```